File: Practica 3/cola.java
public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = -1;
    }

    public void insert(long e) {
        if (!isFull()) {
            fin = (fin + 1) % n;
            arreglo[fin] = e;
        }
    }

    public long remove() {
        if (!isEmpty()) {
            long e = arreglo[inicio];
            inicio = (inicio + 1) % n;
            return e;
        }
        return -1; 
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[inicio];
        }
        return -1; 
    }

    public boolean isEmpty() {
        return inicio == (fin + 1) % n;
    }

    public boolean isFull() {
        return inicio == (fin + 2) % n;
    }

    public int size() {
        return (fin - inicio + 1 + n) % n;
    }
}


public class Main {
    public static void main(String[] args) {
        Cola cola = new Cola(5);
        cola.insert(1);
        cola.insert(2);
        System.out.println(cola.remove()); 
        System.out.println(cola.peek());  
    
}

File: Practica 3/cola.py
class Cola:
    def __init__(self, n):
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = -1
        self.n = n

    def insert(self, e):
        if not self.isFull():
            self.fin = (self.fin + 1) % self.n
            self.arreglo[self.fin] = e

    def remove(self):
        if not self.isEmpty():
            e = self.arreglo[self.inicio]
            self.inicio = (self.inicio + 1) % self.n
            return e

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.inicio]

    def isEmpty(self):
        return self.inicio == (self.fin + 1) % self.n

    def isFull(self):
        return self.inicio == (self.fin + 2) % self.n

    def size(self):
        return (self.fin - self.inicio + 1 + self.n) % self.n


cola = Cola(5)
cola.insert(1)
cola.insert(2)
print(cola.remove()) 
print(cola.peek()) 
